src/common/lock.ts

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+export class Lock {
```

Comment:
Suggest to either use a less generic name or add some comments about the behavior. i.e. `execute` is aborted immediately if the it's locked instead of waiting until unlock.